pkg/cloud/services/compute/availabilityzone.go

```diff
@@ -30,3 +30,21 @@ func (s *Service) GetAvailabilityZones() ([]availabilityzones.AvailabilityZone,
 
 	return availabilityZoneList, nil
 }
+
+// GetAvailableAvailabilityZones does as it says, but only returns usable AZs.
+func (s *Service) GetAvailableAvailabilityZones() ([]availabilityzones.AvailabilityZone, error) {
+        availabilityZoneList, err := s.getComputeClient().ListAvailabilityZones()
```

Comment:
Maybe slightly better to use the function above. Just a suggestion
```suggestion
        availabilityZoneList, err := s.GetAvailabilityZones()
```

Reply:
Followed Matt's advice below.